pkg/clusteragent/admission/mutate/config/filter.go

```diff
@@ -0,0 +1,31 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build kubeapiserver
+
+package config
+
+import (
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	mutatecommon "github.com/DataDog/datadog-agent/pkg/clusteragent/admission/mutate/common"
+)
+
+type FilterConfig struct {
```

Comment:
I'm putting down my thoughts on the structure here having some more time to think about the `FilterConfig`... 

- Each of the `Config` structs is currently public with private fields. That means either folks can construct an empty `FilterConfig{}`, or they can use `NewFilterConfig`-- but really there's only one way of constructing them. Are they private for internal testing? We're not doing that though.

- If the fields aren't meant to ever be touched from the outside, why is this function being exposed this way?

--- 

TLDR: maybe its naming? 😭 

- Rename `NewFilter()` -> `MutationFilter` (for consistency with what it does).
- Rename `NewFilterConfig()` -> `NewMutationFilterConfig()`

No pressure to change anything.